pkg/config/configsections/container.go

```diff
@@ -18,8 +18,82 @@ package configsections
 
 import (
 	"fmt"
+	"time"
+
+	"github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+	"github.com/test-network-function/test-network-function/pkg/tnf/interactive"
+)
+
+const (
+	defaultTimeoutSeconds = 10
+)
+
+var (
+	expectersVerboseModeEnabled = false
+	// DefaultTimeout for creating new interactive sessions (oc, ssh, tty)
+	DefaultTimeout = time.Duration(defaultTimeoutSeconds) * time.Second
 )
 
+// Container is a construct which follows the Container design pattern.  Essentially, a Container holds the
+// pertinent information to perform a test against or using an Operating System Container.  This includes facets such
+// as the reference to the interactive.Oc instance, the reference to the test configuration, and the default network
+// IP address.
+type Container struct {
+	ContainerConfig
+	Oc                      *interactive.Oc
+	DefaultNetworkIPAddress string `yaml:"defaultnetworkipaddress" json:"defaultnetworkipaddress"`
+}
+
+// Helper used to instantiate an OpenShift Client Session.
+func GetOcSession(pod, container, namespace string, timeout time.Duration, options ...interactive.Option) *interactive.Oc {
+	// Spawn an interactive OC shell using a goroutine (needed to avoid cross expect.Expecter interaction).  Extract the
+	// Oc reference from the goroutine through a channel.  Performs basic sanity checking that the Oc session is set up
+	// correctly.
+	var containerOc *interactive.Oc
+	ocChan := make(chan *interactive.Oc)
+
+	goExpectSpawner := interactive.NewGoExpectSpawner()
+	var spawner interactive.Spawner = goExpectSpawner
+
+	go func() {
+		oc, outCh, err := interactive.SpawnOc(&spawner, pod, container, namespace, timeout, options...)
+		gomega.Expect(outCh).ToNot(gomega.BeNil())
+		gomega.Expect(err).To(gomega.BeNil())
+		// Set up a go routine which reads from the error channel
+		go func() {
+			log.Debugf("start watching the session with container %s/%s", oc.GetPodName(), oc.GetPodContainerName())
+			select {
+			case err := <-outCh:
+				log.Fatalf("OC session to container %s/%s is broken due to: %v, aborting the test run", oc.GetPodName(), oc.GetPodContainerName(), err)
+			case <-oc.GetDoneChannel():
+				log.Debugf("stop watching the session with container %s/%s", oc.GetPodName(), oc.GetPodContainerName())
+			}
+		}()
+		ocChan <- oc
+	}()
+
+	containerOc = <-ocChan
```

Comment:
Why do the above section if you are just blocking waiting for the interactive session to come back?

Reply:
Fyi, I did not modify this code in this PR, just moved the code to resolve dependencies. I believe the reason for this section is to return the "oc" context right away and to gracefully handles normal termination and error cases. The interactive session is long lived and multiples command are ran on it. This section monitors the status in the background

Reply:
Yeah I saw that it was moved.  Interesting function I'll have to dig into it some more.